### PR TITLE
Heap: add additional params

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> add `timestamp` and `idempotency_key` params

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/heap/transform.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ function responseBuilderSimple(message, category, destination) {
   if (payload) {
     if (payload.properties) {
       payload.properties = flattenJson(payload.properties);
+      // remove duplicate key as it is being passed at root.
+      if (payload.properties.idempotencyKey) {
```
Comment (ItsSudip) [inline v0/destinations/heap/transform.js:20]:
```suggestion
      if (get(payload,"properties.idempotencyKey")) {
```

Comment (saikumarrs) [reply v0/destinations/heap/transform.js:20]: Even better,
`payload?.properties?.idempotencyKey`

Comment (akashrpo) [reply v0/destinations/heap/transform.js:20]: IMO it is not required as we already have `payload.properties` check above.

Comment (saikumarrs) [reply v0/destinations/heap/transform.js:20]: What are the expected outcomes of the `flattenJson` method?

Comment (akashrpo) [reply v0/destinations/heap/transform.js:20]: converts nested layers into only one layer of key/value pair.

Comment (saikumarrs) [reply v0/destinations/heap/transform.js:20]: Can it return `undefined` or `null` as a result of some internal error?

Comment (akashrpo) [reply v0/destinations/heap/transform.js:20]: No

**Review by ItsSudip — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**